EIPS/eip-4824.md

```diff
@@ -0,0 +1,288 @@
+mapping (address => bool) daoRegistered;
+		address offeredAddress;
+
+		event TransferOffered(address currentOwner, address newOwner);
+		event TransferAccepted(address oldOwner, address newOwner);
+
+		function initialize(address _daoAddress) {
+			if (_daoAddress == msg.sender) {
+				daoAddress = _daoAddress;
+				emit TransferAccepted(address(0), _daoAddress);
+			} else {
+				offeredAddress = _daoAddress;
+				emit TransferOffered(address(0), _daoAddress);
+			}
+
+		}
+
+		function daoURI() external view returns (string _daoURI) {
+			return daoURI;
+		}
+		
+    function offerTransfer(address _offeredAddress) external {
+			if (msg.sender != daoAddress) revert NotOwner();
+			offeredAddress = _offeredAddress;
+
+			emit TransferOffered(daoAddress, _offeredAddress);
+		}
+
+    function acceptTransfer() external {
+			address _oldOwner = daoAddress;
+			if (msg.sender != offeredAddress) revert NotOffered();
+			daoAddress = msg.sender;
+			emit TransferAccepted(_oldOwner, daoAddress);
+		}
+}
+```
+
+If a DAO uses an external registration contract, the DAO SHOULD use a common registration factory contract to enable efficient network indexing.
+
+```solidity
+pragma solidity ^0.8.1;
+
+/// @title EIP-4824 DAOs
+/// @dev See <https://eips.ethereum.org/EIPS/eip-4824>
+
+interface EIP4824RegistrationFactory is CloneFactory {
+    function summonRegistration();
+}
+```
+
+### Members
+
+Members JSON-LD Schema.
+
+```json
+{
+  "@context": "<http://www.daostar.org/schemas>",
+  "type": "DAO",
+  "name": "<name of the DAO>",
+  "members": [
+    {
+      "type": "EthereumAddress",
+      "id": "<address or other identifier>"
+    },
+    {
+      "type": "EthereumAddress",
+      "id": "<address or other identifier>"
+    }
+  ]
+}
+```
+
+### Proposals
+
+Proposals JSON-LD Schema. Every contract implementing this EIP should implement a proposalsURI pointing to a JSON object satisfying this schema.
+
+In particular, any on-chain proposal MUST be associated to an id of the form CAIP10_ADDRESS + “?proposalId=” + PROPOSAL_COUNTER, where CAIP10_ADDRESS is an address following the CAIP-10 standard and PROPOSAL_COUNTER is an arbitrary identifier such as a uint256 counter or a hash that is locally unique per CAIP-10 address. Off-chain proposals MAY use a similar id format where CAIP10_ADDRESS is replaced with an appropriate URI or URL.
+
+```json
+{
+  "@context": "http://www.daostar.org/schemas",
+  "type": "DAO",
+  "name": "<name of the DAO>",
+  "proposals": [
+    {
+      "type": "proposal",
+      "id": "<proposal ID>",
+      "name": "<name or title of proposal>",
+      "contentURI": "<URI to content or discussion>",
+      "status": "<status of proposal>",
+      "calls": [
+        {
+	  "type": "CallDataEVM",
+	  "operation": "<call or delegate call>",
+          "from": "<EthereumAddress>",
+          "to": "<EthereumAddress>",
+	  "value": "<value>",
+          "data": "<call data>"
+        }
+      ]
+    }
+  ]
+}
```

Comment:
Note: I *do* support at least internal consistency within a given document.  This file is a mix of tabs and spaces that at least should be addressed, even if the author prefers their other stylistic choices.

Reply:
fixed; changed everything to 4 spaces. since JSON style is not a blocker, I reverted the [ { line format to the one we've been using throughout our docs. I find it slightly preferable